react/src/components/Register/Register.js

```diff
@@ -1,29 +1,106 @@
 import React, { useState } from "react";
 import "../genericStyles.css";
 
-const Register = ({ changeStatus, addUserToDatabase, errorMessage }) => {
+const Register = ({ changeStatus }) => {
   const [email, setEmail] = useState("");
   const [username, setUsername] = useState("");
   const [password, setPassword] = useState("");
+  const [avatar, setAvatar] = useState(null);
+  const [emailError, setEmailError] = useState("");
+  const [usernameError, setUsernameError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
 
   const _onEmailChange = (event) => {
     setEmail(event.target.value);
+    setEmailError("");
   };
 
   const _onUsernameChange = (event) => {
     setUsername(event.target.value);
+    setUsernameError("");
   };
 
   const _onPasswordChange = (event) => {
-    setPassword(event.target.value);
+    const value = event.target.value;
+    setPassword(value);
+
+    const regex = /^(?=.*[a-zA-Z])(?=.*\d)[A-Za-z\d]{8,20}$/;
+    if (value === "") {
+      setPasswordError("Password is required.");
+    } else if (!regex.test(value)) {
+      setPasswordError(
+        "Your password must be 8-20 characters long, contain letters and numbers, and must not contain spaces, special characters, or emoji."
+      );
+    } else {
+      setPasswordError("");
+    }
+  };
+
+  const _onAvatarChange = (event) => {
```

Comment:
I'm not sure this is gonna work..the setAvatar needs a string in the form of an image src, so it needs something like: http://localhost:8000/media/${avatarIcon} where avatarIcon is "avatars/icon.png"

Reply:
maybe you need to create a new variable if you need the event.target.files[0] for a reason